### PR TITLE
Terms needed for openMINDS/ephys

### PR DESCRIPTION
**Pull request by apdavison**
adds the following terms, together with some instances for each:

- PatchClampVariation
- CranialWindowType
- CultureType
- MeasuredQuantity
- AnatomicalPlane

For where these terms are used, see https://github.com/HumanBrainProject/openMINDS_ephys

### CODE REVIEW

File: instances/cellCultureType/primary.jsonld
Comment (lzehl) [inline instances/cellCultureType/primary.jsonld]: should be null

File: instances/cellCultureType/secondary.jsonld
Comment (lzehl) [inline instances/cellCultureType/secondary.jsonld]: should be null

File: instances/cranialWindowType/crystalline.jsonld
Comment (lzehl) [inline instances/cranialWindowType/crystalline.jsonld]: should be null

File: instances/cranialWindowType/glassCoverslip.jsonld
Comment (lzehl) [inline instances/cranialWindowType/glassCoverslip.jsonld]: should be null

File: instances/cranialWindowType/grinLens.jsonld
Comment (lzehl) [inline instances/cranialWindowType/grinLens.jsonld]: should be null

File: instances/cranialWindowType/polymer.jsonld
Comment (lzehl) [inline instances/cranialWindowType/polymer.jsonld]: should be null

File: instances/cranialWindowType/prism.jsonld
Comment (lzehl) [inline instances/cranialWindowType/prism.jsonld]: should be null

File: instances/cranialWindowType/reinforcedThinnedSkullTranscranial.jsonld
Comment (lzehl) [inline instances/cranialWindowType/reinforcedThinnedSkullTranscranial.jsonld]: should be null

File: instances/cranialWindowType/thinnedSkullTranscranial.jsonld
Comment (lzehl) [inline instances/cranialWindowType/thinnedSkullTranscranial.jsonld]: should be null

File: instances/cranialWindowType/transcranial.jsonld
Comment (lzehl) [inline instances/cranialWindowType/transcranial.jsonld]: should be null

File: instances/measuredQuantity/chlorideReversalPotential.jsonld
Comment (lzehl) [inline instances/measuredQuantity/chlorideReversalPotential.jsonld]: should be null

File: instances/measuredQuantity/compensationCurrent.jsonld
Comment (lzehl) [inline instances/measuredQuantity/compensationCurrent.jsonld]: should be null

File: instances/measuredQuantity/inputResistance.jsonld
Comment (lzehl) [inline instances/measuredQuantity/inputResistance.jsonld]: should be null

File: instances/measuredQuantity/liquidJunctionPotential.jsonld
Comment (lzehl) [inline instances/measuredQuantity/liquidJunctionPotential.jsonld]: should be null

File: instances/measuredQuantity/measuredHoldingPotential.jsonld
Comment (lzehl) [inline instances/measuredQuantity/measuredHoldingPotential.jsonld]: should be null

File: instances/measuredQuantity/measuredHoldingPotential.jsonld
Comment (lzehl) [inline instances/measuredQuantity/measuredHoldingPotential.jsonld:11]: why measured? why not only "holding potential"?

File: instances/measuredQuantity/membranePotential.jsonld
Comment (lzehl) [inline instances/measuredQuantity/membranePotential.jsonld]: should be null

File: instances/measuredQuantity/sealResistance.jsonld
Comment (lzehl) [inline instances/measuredQuantity/sealResistance.jsonld]: should be null

File: instances/measuredQuantity/seriesResistance.jsonld
Comment (lzehl) [inline instances/measuredQuantity/seriesResistance.jsonld]: should be null

File: instances/patchClampVariation/cellAttachedPatch.jsonld
Comment (lzehl) [inline instances/patchClampVariation/cellAttachedPatch.jsonld:2]: these is a technique... I do not think that we should capture it again. I do see why you only want to separate the patch clamp techniques but we should find a better way for this.

Any ideas?

Comment (lzehl) [reply instances/patchClampVariation/cellAttachedPatch.jsonld:2]: this applies for patchClampVariations you add. Other way around, but possible rejected by the EBRAINS curation team: reducing technique to "patch clamp" and leaving all variations in here. Does this work? I can find a solution for the curation team (e.g. by allowing not only technique but also the patch variations) for technique properties.

File: schemas/cellCultureType.schema.tpl.json
Comment (lzehl) [inline schemas/cellCultureType.schema.tpl.json]: cellCultureType or you need to rename the file

File: schemas/measuredQuantity.schema.tpl.json
Comment (lzehl) [inline schemas/measuredQuantity.schema.tpl.json:2]: could we add also "local field potential" etc here? or would you keep those separate for example under a terminology named "neuronal signal" ?

**Review by lzehl — CHANGES_REQUESTED** [review]
@apdavison this is a great PR (technically only a minor issue with descriptions being an empty string instead of null) 

I just have some questions that need to be resolved. This will affect other parts of openMINDS.

File: instances/patchClampVariation/cellAttachedPatch.jsonld
Comment (apdavison) [reply instances/patchClampVariation/cellAttachedPatch.jsonld:2]: We could add them as techniques, but then how do we limit the field to only a specific subset of instances?

File: schemas/measuredQuantity.schema.tpl.json
Comment (apdavison) [reply schemas/measuredQuantity.schema.tpl.json:2]: In the schema we've developed here, "local field potential" would be a recording, not a measurement, but maybe the distinction isn't important.

It occurs to me we could perhaps add this information to the Channel schema.

File: instances/measuredQuantity/measuredHoldingPotential.jsonld
Comment (apdavison) [reply instances/measuredQuantity/measuredHoldingPotential.jsonld:11]: because we have "targetHoldingPotential" and "measuredHoldingPotential". Maybe not essential to have "measured" in the name here since "measured" is in the name of the controlled term type, but it seemed safest to keep it.

File: instances/patchClampVariation/cellAttachedPatch.jsonld
Comment (lzehl) [reply instances/patchClampVariation/cellAttachedPatch.jsonld:2]: exactly for this we do not have yet any functionality...

I think we should do it the other way around. I will delete the more specific patch clamp techniques we have from the techniques terminology and we leave them in patchClampVariations. For the technique properties I will add then the patchClampVariations as additional linkedType so that users can immediately state that they are doing "whole cell patch clamp". does this make sense?

File: schemas/cellCultureType.schema.tpl.json
Comment (apdavison) [reply schemas/cellCultureType.schema.tpl.json]: done

File: schemas/measuredQuantity.schema.tpl.json
Comment (lzehl) [reply schemas/measuredQuantity.schema.tpl.json:2]: measuredQuantity sounds like a general terminology. why would you differentiate recording from it? for me a recording is also a measurement, no? But I understand if you want to differentiate between extracellular measurements and intracellular measurements OR do you want to differentiate between scalar values and timeseries?

File: instances/patchClampVariation/cellAttachedPatch.jsonld
Comment (lzehl) [reply instances/patchClampVariation/cellAttachedPatch.jsonld:2]: perfect :) this issue might occur for future extensions (and other parts of ephys) as well.

File: schemas/measuredQuantity.schema.tpl.json
Comment (apdavison) [reply schemas/measuredQuantity.schema.tpl.json:2]: the latter - we've chosen to represent a measurement at a single point in time as a Measurement (scalar - stored directly in KG), a sequence of measurements as a Recording (timeseries - stored in a File)

Comment (apdavison) [reply schemas/measuredQuantity.schema.tpl.json:2]: but I agree it doesn't make sense to distinguish MeasuredQuantity and RecordedQuantity, so we should use MeasuredQuantity for both Measurement and Recording/Channel.

Comment (lzehl) [reply schemas/measuredQuantity.schema.tpl.json:2]: okay

**Comment by lzehl** [comment]
@apdavison is this ready for a second review?

**Comment by apdavison** [comment]
@lzehl yes

File: instances/measuredQuantity/measuredHoldingPotential.jsonld
Comment (lzehl) [reply instances/measuredQuantity/measuredHoldingPotential.jsonld:11]: I see.

**Review by lzehl — COMMENTED** [review]
@apdavison I forgot this the last time, but you should add the new overall terminologies to the controlledTerms/terminology/*.jsonld instances and add to the schemas the "studyTarget" category for the once that make sense as a studyTarget (e.g., cellCultureType and measuredQuantity)

otherwise it looks good. do you want to wait for @tgbugs review as well?

**Comment by apdavison** [comment]
@lzehl I think this is fixed now. Please could you give it another check, and merge if everything looks ok?

**Review by lzehl — CHANGES_REQUESTED** [review]
@apdavison thank you for the corrections. There is only one thing left: could you add a new line to all files that are missing one?

**Review by lzehl — APPROVED** [review]

**Pull request merged**
